cpp/src/arrow/compute/kernels/vector_replace_test.cc

```diff
@@ -223,6 +223,22 @@ TYPED_TEST(TestReplaceNumeric, ReplaceWithMask) {
                this->mask("[false, false, null, null, true, true]"),
                this->array("[10, null]"),
                this->array("[null, null, null, null, 10, null]"));
+  this->Assert(ReplaceWithMask, this->array("[1, null, 1]"),
+               this->mask("[false, true, false]"),
+               this->array("[7]"),
+               this->array("[1, 7, 1]"));
+  this->Assert(ReplaceWithMask, this->array("[1, null, 1, 7]"),
+               this->mask("[false, true, false, true]"),
+               this->array("[7, 20]"),
+               this->array("[1, 7, 1, 20]"));
+  this->Assert(ReplaceWithMask, this->array("[1, 2, 3, 4]"),
+               this->mask("[false, true, false, true]"),
+               this->array("[null, null]"),
+               this->array("[1, null, 3, null]"));
+  this->Assert(ReplaceWithMask, this->array("[null, 2, 3, 4]"),
+               this->mask("[true, true, false, true]"),
+               this->array("[1, null, null]"),
```

Comment:
I would suggest to move all these new tests to a new test function to clarify the purpose of this PR.

TYPED_TEST(TestReplaceNumeric, `ReplaceWithMaskForNullValuesAndMaskEnabled`)

Reply:
solved